client/src/components/reviews/SortSelector.jsx

```diff
@@ -0,0 +1,66 @@
+import React, { useState } from 'react';
+import {
+  Select, MenuItem, FormControl, InputLabel
+} from '@material-ui/core';
+import { makeStyles } from '@material-ui/core/styles';
+
+const useStyles = makeStyles((theme) => ({
+  formControl: {
+    margin: theme.spacing(1),
+    minWidth: 170
+  },
+  selectEmpty: {
+    marginTop: theme.spacing(2)
+  }
+}));
+
+const SortSelector = ({
+  setSortValue, setReviewResults, setReviewTileList,
+  setReviews, setPageState, setTileCount }) => {
+  const [selectText, setSelectText] = useState('');
+
+  const handleChange = (event) => {
+    const promise = new Promise((resolve) => {
+      setSelectText(event.target.value);
+      resolve(selectText);
+    });
+    promise
+      .then(() => {
+        setSortValue(event.target.value);
+      }).then(() => {
+        setReviewTileList([]);
+        setReviews([]);
+        setReviewResults([]);
+        setTileCount(2);
+      }).then(() => {
+        // This will trigger getAllReviewResults
+        setPageState(1);
```

Comment:
Why are there multiple thens here? This appears like it could be done in the parent class if you watch the selectText value

Reply:
Yeah, I could probably combine the first and second .then. I would at least need one additional .then so that the review list could empty out before the new order loads.